### ISSUE TITLE
iOS extension option custom tag code injection inserts nothing

### ISSUE THREAD

**Issue by arashmoknat** [YoYoGames/GameMaker-Bugs#6059]
### Description

In the code injection part of an iOS extension, putting in a custom tag that inserts an extension user option isn't working.
Example:
```
<MyCustomTag>
${YYEXTOPT_MyExtension_Option}
</MyCustomTag>
```
The corresponding tags in the .mm file would be like:
`NSString* myExtensionOption = @"${MyCustomTag}";`
which turns into:
`NSString* myExtensionOption = @"";`
on build.

`${YYExtAppDelegateIncludes}` header also seems to go missing.

[CodeInjectionExtensionOptionBug.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/15505936/CodeInjectionExtensionOptionBug.zip)

### Expected Change

`NSString* myExtensionOption = @"<Option value here>"`;

### Steps To Reproduce

1. Start GameMaker
2. Open attached project
3. Build to iOS
4. See the issue

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.4.1 (Monthly)

### Which operating system(s) are you seeing the problem on?

Windows 10

### Which platform(s) are you having trouble building this project for?

iOS

### Are you running GameMaker from inside your Steam library?

No

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by arashmoknat** [comment]
Ok, so I made a mistake, I was going through the manual again and the custom tag must start with YYIos. Changing the tag to that fixed the issue, but the other thing I said about the header file ${YYExtAppDelegateIncludes} missing does still happen.

**Comment by arashmoknat** [comment]
Also, right now even if I pack the code injection tags together, for example `<YYIocCustomTag>${YYEXTOPT_MyExt_CustomOption}</YYIocCustomTag>` the resulting code inserts a new line at the end anyway, which in something like the provided example will break the string definition.

**Comment by DiasFranciscoA** [comment]
Hello @arashmoknat,

Thank you for reaching out to us. It appears that your approach to code injection is not aligned with the intended usage. If your goal is to retrieve the extension option value at runtime, code injection is not necessary. Instead, you should use the method demonstrated in our AdMob extension to access extension option values at runtime:

```objc
extOptGetString("extensionName", "optionName");
```

Please refer to our [AdMob Extension for iOS](https://github.com/YoYoGames/GMEXT-AdMob/blob/main/source/AdMob_gml/extensions/admob/iOSSource/GoogleMobileAdsGM.mm#L23) to see how this method is implemented.

**Comment by arashmoknat** [comment]
I see, there isn't much documentation about the native side of making extensions so I wasn't even aware of that function.
One of the issues mentioned is still relevant though. The missing '${YYExtAppDelegateIncludes}' includes. It's easy enough to work around though.

**Comment by Emc1923** [comment]
Verified fixed as of  IDE v2024.6.0.157  Runtime v2024.6.0.205.
